### ISSUE TITLE
Return OBO MM configuration to use Primary Activity as the Primary Directory vocabulary

### ISSUE THREAD

**Issue by ColmDC** [DigitalCommons/mykomap#235]
## Description
Return OBO MM configuration to use Primary Activity as the Primary Directory vocabulary.

## Acceptance Criteria

**Comment by ColmDC** [comment]
Was there a problem with using Primary Activities for the Primary Directory?

**Comment by lin-d-hop** [comment]
This is cryptic to an outsider. I wonder if we can add any supportive detail, or link to anywhere in which this jargon is explained?

**Comment by wu-lee** [comment]
(Is this specifically a Mykomap issue?  Seems like this is in the wrong project - the work for this was done as a part of this issue https://github.com/DigitalCommons/owned-by-oxford-project/issues/26)

Just to be clear, I'm referring to this map: https://dev.maps.solidarityeconomy.coop/obo-public/

There the "Nature Of" field is being used on the directory panel. IIRC we were asked to use that as the first field in the filter list (see issue 26 above) because they wanted it to be the first one people see. And the consequence of that is that it is used for the directory panel. I think this was mentioned in the google documents I tried to summarise in that issue 26, although I can't find this mentioned in the issue specifically... I also recall pointing out this would be a consequence, and it would look strange because that field's taxonomy (AKA vocab) has only two values.

As an aside - we were also asked to combine "Primary Activity" and "Secondary Activities" into a single field, so users can search for them as one thing.  This field happens to be called "Combined Activities", and of course allows multiple values in the field. If we do want to show activities instead, possibly that deserves to be in the directory rather than "Primary Activity".  (Multi-value fields will be a problem for pin icons/colours however.)

**Comment by ColmDC** [comment]
> (Is this specifically a Mykomap issue? Seems like this is in the wrong project - the work for this was done as a part of this issue [DigitalCommons/owned-by-oxford-project#26](https://github.com/DigitalCommons/owned-by-oxford-project/issues/26))

There is a motivation to move away from having as many repositories as we have. So if an issue is a generic MM issue, even if it arises through a specific project, we can manage it as an issue in the main repository. But this issue does seem to be very local to the OBO project. At some point we will not need repositories for each project and we will need to have a different protocol for tagging project specific issues.

**Comment by ColmDC** [comment]
@lin-d-hop we will just clarify the issue with @nick in the comments and then rephrase the Issue description. 

> IIRC we were asked to use that as the first field in the filter list (see issue 26 above) because they wanted it to be the first one people see. And the consequence of that is that it is also the one used for the directory panel (a design choice made in Mykomap's deep history).

I recall now. @wu-lee Can you recall if that connection between the first filter item and choice of Primary directory category would be hard to seperate?

**Comment by ColmDC** [comment]
>This field happens to be called "Combined Activities", and of course allows multiple values in the field. If we do want to show activities instead, possibly that deserves to be in the directory rather than "Primary Activity". (Multi-value fields will be a problem for pin icons/colours however.)

If we did this, I can't see why there would need to be any changes to the use of colours? Pin colour remains hardwired to Primary Activity? But what we will loose is the colour key, as in the old map the key was provided by a coloured bar beside the name of the Primary Activiity in the Directory Panel. 
![Owned-by-Oxford](https://github.com/DigitalCommons/mykomap/assets/108667445/fe685a3b-45e7-470e-8337-1c731689fbd8)

We also do not have that implicit key to the pin colours if we have Nature of Organisation in the Directory panel.

![Owned-by-Oxford-Nature](https://github.com/DigitalCommons/mykomap/assets/108667445/b5d62296-6a11-4f64-8f74-b4cb772d3406)

**Comment by wu-lee** [comment]
> If we did this, I can't see why there would need to be any changes to the use of colours? Pin colour remains hardwired to Primary Activity? But what we will loose is the colour key, as in the old map the key was provided by a coloured bar beside the name of the Primary Activiity in the Directory Panel.

Agreed, pins would still be coloured as you say, but we'd lose colours on the directory if it weren't a single-valued economic activity.

> We also do not have that implicit key to the pin colours if we have Nature of Organisation in the Directory panel.

Agreed.

> Can you recall if that connection between the first filter item and choice of Primary directory category would be hard to seperate?

Not that hard, but it would need inventing some alternative way to define it. Possibly some new field in the config, to allow backward compatibility and a way to specify this option in the URL parameters easily; alternatively perhaps allowing the things in the `filterableFields` list to be objects in the general case, which might also allow for more flexibility in future, e.g. setting other values like the default filter value, which is something else we have an issue for (although not one I can currently locate). At the expense of ease of URL-parametering. (I currently tend towards the latter, URL parameters are problematic anyway.)

**Comment by ColmDC** [comment]
"

    Can you recall if that connection between the first filter item and choice of Primary directory category would be hard to seperate?

Not that hard, but it would need inventing some alternative way to define it. Possibly some new field in the config, to allow backward compatibility and a way to specify this option in the URL parameters easily; alternatively perhaps allowing the things in the filterableFields list to be objects in the general case, which might also allow for more flexibility in future, e.g. setting other values like the default filter value, which is something else we have an issue for (although not one I can currently locate). At the expense of ease of URL-parametering. (I currently tend towards the latter, URL parameters are problematic anyway.)"

So I think this requires a new ticket on its own. Let's move this ticket back down the list until this new issue is completed.

**Comment by ColmDC** [comment]
Return to this when https://github.com/DigitalCommons/mykomap/issues/237 has been implemented.

**Comment by ColmDC** [comment]
I've tagged is as blocked, but I think its requirements are no clear (and easy once https://github.com/DigitalCommons/mykomap/issues/237 is implemented) so moving to Backlog.

**Comment by ColmDC** [comment]
> I've tagged is as blocked, but I think its requirements are no clear (and easy once [#237](https://github.com/DigitalCommons/mykomap/issues/237) is implemented) so moving to Backlog.

[#237](https://github.com/DigitalCommons/mykomap/issues/237) is now implemented, so it should be just a matter of changing the configurations.
